scripts/file_collector.py

```diff
@@ -0,0 +1,187 @@
+        variables["spatial"],
+        variables["percentile"],
+        area,
+        date_range,
+        ensemble,
+    )
+
+    logger.info("Collecting models")
+    models = get_models(sesh, variables["percentile"], ensemble)
+
+    var_name = "_".join(
+        [
+            variables["variable"],
+            variables["time_of_year"],
+            variables["temporal"],
+            variables["spatial"],
+            variables["percentile"],
+        ]
+    )
+
+    logger.info("Fetching data for {}".format(var_name))
+
+    paths = []
+    [paths.extend(query_backend(sesh, model, query_args)) for model in models]
+    logger.info("Writing file path for {} to file".format(var_name))
+
+    with open("file_paths.txt", "a") as file_:
+        for path in paths:
+            file_.write(path + "\n")
+
+
+def query_backend(sesh, model, query_args):
+    """Return the desired file name for a particular climate model"""
+    logger.debug("Running query_backend() with args: %s, %s", model, query_args)
+    ids = [
+        search_for_unique_ids(
+            sesh,
+            ensemble_name=query_args["ensemble_name"],
+            model=model,
+            emission=query_args["emission"],
+            time=query_args["time"],
+            variable=var,
+            timescale=query_args["timescale"],
+            cell_method=query_args["cell_method"],
+        )
+        for var in query_args["variable"]
+    ]
+    file_names = []
+    for id_ in ids:
+        for model_id in list(id_):
+            data_file = (
+                sesh.query(DataFile).filter(DataFile.unique_id == model_id).one()
+            )
+            if data_file is not None:
+                file_names.append(data_file.filename)
+    return file_names
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument(
+        "-c", "--csv", help="CSV file containing rules", default="./data/rules.csv",
+    )
+    parser.add_argument(
+        "-d",
+        "--date-range",
+        help="30 year period for data",
+        choices=["2020", "2050", "2080"],
+        default="2080",
+    )
+    parser.add_argument(
+        "-r", "--region", help="Selected region", default="bc", choices=REGIONS.keys()
+    )
+    parser.add_argument(
+        "-u",
+        "--url",
+        help="Geoserver URL",
+        default="http://docker-dev01.pcic.uvic.ca:30123/geoserver/bc_regions/ows",
+    )
+    parser.add_argument(
+        "-x",
+        "--connection-string",
+        help="Database connection string",
+        default="postgres://ce_meta_ro@db3.pcic.uvic.ca/ce_meta_12f290b63791",
+    )
+    parser.add_argument(
+        "-e",
+        "--ensemble",
+        help="Ensemble name filter for data files",
+        default="p2a_rules",
+    )
+    args = parser.parse_args()
+    region = get_region(args.region, args.url)
+
+    file_collection(
+        args.csv, args.date_range, region, args.ensemble, args.connection_string
+    )
```

Comment:
We should consider adding an option to return the result directly to `stdout` like [so](https://github.com/pacificclimate/crmprtd/blob/master/crmprtd/download.py#L163). This would allow us to pipe the result directly into the next script without having to create a file.